### PR TITLE
INFINITY-2040 Developer-facing changes to YAML port handling

### PR DESCRIPTION
**Pull request by nickbp**
Updates scheduler code and docs to reflect the above changes, and updates per-service yamls to retain their current behavior given the changes to default behavior.

Changes:
- Removes deprecated fields from VIPs (previously threw helpful error when provided)
- Adds 'advertise' setting to ports.
- Changes default endpoint listing behavior: Developer must manually set the newly-added `advertise: true` setting to ports which should be listed in the `endpoints` command. Previously, this was determined only by whether a VIP was present.
- Changes default env behavior: Developer must manually provide an `env-key` if they want the port to be advertised in task envs, when previously a default of `PORT_<PORT_NAME_HERE>` was provided when `env-key` was unset. Developers who didn't provide an `env-key` and used the default `PORT_<NAME_HERE>` envvar will now need to specify `env-key` explicitly.

### CODE REVIEW

File: docs/pages/reference/yaml-reference.md [hunk 1]
```diff
@@ -252,9 +254,11 @@ This documentation effectively reflects the Java object tree under [RawServiceSp
 
       * `env-key`
 
-        This may be used to customize the environment variable used to advertise this port within the task.
+        This may be used to customize the environment variable used to advertise this port within the task. This is most useful when a random dynamic port is being used, as it allows the task to know what port was allocated for it.
```
Comment (mrbrowning) [inline docs/pages/reference/yaml-reference.md]: "Customize" probably isn't the right word anymore, now that the environment variable won't be there if `env-key` is unset.

File: docs/pages/tutorials/kafka-tutorial.md [hunk 2]
```diff
@@ -17,11 +17,13 @@ In this tutorial we'll be walking through step-by-step instructions on getting a
 
 4. The name of a task can be anything but since Kafka has brokers, change the type of the task in the Kafka pod from `node` to `broker`. Also, notice that `count` is specified as a property of the `kafka` pod. This indicates there will be `count` kafka pods, each of which will run a `broker`.
 
-5. Next, add the following property to the `broker` task in order to open up the necessary port.
+5. Next, add the following property to the `broker` task in order to open up the necessary port, and mark it as `advertise`d so that it appears in service information.
```
Comment (mrbrowning) [inline docs/pages/tutorials/kafka-tutorial.md]: Nit: "service endpoint information"?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/NamedVIPEvaluationStage.java [hunk 3]
```diff
@@ -21,9 +21,8 @@
     public NamedVIPEvaluationStage(
             NamedVIPSpec namedVIPSpec,
             String taskName,
-            Optional<String> resourceId,
-            boolean useDefaultExecutor) {
-        super(namedVIPSpec, taskName, resourceId, useDefaultExecutor);
+            Optional<String> resourceId) {
+        super(namedVIPSpec, taskName, resourceId);
```
Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/NamedVIPEvaluationStage.java]: Looks like you'll need to rebase. I'm getting rid of these arguments, but that'll be in later today or tomorrow.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPort.java [hunk 4]
```diff
@@ -28,6 +31,10 @@ public String getEnvKey() {
         return envKey;
     }
 
+    public boolean isAdvertise() {
```
Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPort.java]: Subjective ultra-nit, but `doesAdvertise` for enhanced grammatical compliance?

Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPort.java]: Yeah I was leaning towards java convention here. Maybe isAdvertised to make it less cringe-worthy

File: docs/pages/tutorials/kafka-tutorial.md (hunk 2 above)
Comment (nickbp) [inline docs/pages/tutorials/kafka-tutorial.md]: Done

File: docs/pages/reference/yaml-reference.md (hunk 1 above)
Comment (nickbp) [inline docs/pages/reference/yaml-reference.md]: Good catch, done.

**Comment by nickbp** [comment]
Removed gc_frameworks() as it results in sandbox logs getting wiped. It was originally a workaround for CORE-874.

**Comment by elezar** [comment]
It seems as if the removal of the call to `gc_frameworks` was also attempted in #585 but abandoned.

**Comment by nickbp** [comment]
I remember that commit, it had died in rebase hell independently of the gc_frameworks removal

**Comment by nickbp** [comment]
Bulleted list in PR description covers the changes. Added some more detail there

**Pull request merged**